engine/jobmaster/dm/openapi.go

```diff
@@ -0,0 +1,251 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/pb"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/openapi"
+	dmpkg "github.com/pingcap/tiflow/engine/pkg/dm"
+)
+
+func (jm *JobMaster) initOpenAPI(router *gin.RouterGroup) {
+	wrapper := openapi.ServerInterfaceWrapper{
+		Handler: jm,
+	}
+	// copy from openapi.RegisterHandlersWithOptions
+	router.DELETE("/binlog/tasks/:task-name", wrapper.DMAPIDeleteBinlogOperator)
+
+	router.GET("/binlog/tasks/:task-name", wrapper.DMAPIGetBinlogOperator)
+
+	router.POST("/binlog/tasks/:task-name", wrapper.DMAPISetBinlogOperator)
+
+	router.GET("/config", wrapper.DMAPIGetJobConfig)
+
+	router.PUT("/config", wrapper.DMAPIUpdateJobConfig)
+
+	router.GET("/schema/tasks/:task-name", wrapper.DMAPIGetSchema)
+
+	router.PUT("/schema/tasks/:task-name", wrapper.DMAPISetSchema)
+
+	router.GET("/status", wrapper.DMAPIGetJobStatus)
+
+	router.PUT("/status", wrapper.DMAPIOperateJob)
+}
+
+// DMAPIGetJobStatus implements the api of get job status.
+func (jm *JobMaster) DMAPIGetJobStatus(c *gin.Context, params openapi.DMAPIGetJobStatusParams) {
+	var tasks []string
+	if params.Tasks != nil {
+		tasks = *params.Tasks
+	}
+	resp, err := jm.QueryJobStatus(c.Request.Context(), tasks)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, resp)
+}
+
+// DMAPIGetJobConfig implements the api of get job config.
+func (jm *JobMaster) DMAPIGetJobConfig(c *gin.Context) {
+	cfg, err := jm.GetJobCfg(c.Request.Context())
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	bs, err := cfg.Yaml()
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, string(bs))
+}
+
+// DMAPIUpdateJobConfig implements the api of update job config.
+func (jm *JobMaster) DMAPIUpdateJobConfig(c *gin.Context) {
+	// TODO: support update job config
+}
+
+// DMAPIOperateJob implements the api of operate job.
+func (jm *JobMaster) DMAPIOperateJob(c *gin.Context) {
+	var req openapi.OperateJobRequest
+	if err := c.Bind(&req); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	var op dmpkg.OperateType
+	switch req.Op {
+	case openapi.OperateJobRequestOpPause:
+		op = dmpkg.Pause
+	case openapi.OperateJobRequestOpResume:
+		op = dmpkg.Resume
+	default:
+		_ = c.Error(errors.Errorf("unsupport op type '%s' for operate task", req.Op))
+		return
+	}
+
+	var tasks []string
+	if req.Tasks != nil {
+		tasks = *req.Tasks
+	}
+	err := jm.OperateTask(c.Request.Context(), op, nil, tasks)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.Status(http.StatusOK)
+}
+
+// DMAPIGetBinlogOperator implements the api of get binlog operator.
+func (jm *JobMaster) DMAPIGetBinlogOperator(c *gin.Context, taskName string, params openapi.DMAPIGetBinlogOperatorParams) {
```

Comment:
hope there's not many operators so we don't need to add a `total` field and paging

Reply:
add a todo